### PR TITLE
Feature/make city list view controller UI & mainTabBar, launchScreen 수정

### PR DESCRIPTION
**Pull request by 97JUN**
# 이번 PR에서 한 작업
- launch Screen을 코드로 작성하여 실행하도록 수정.
- AppFont & AppColor 파일을 추가하여 사용할 font, color 를 등록
- 생성했던 MainTabBar 의 firstViewController, secondVieController 수정.
- CityListViewController에 title label, Button, TableView 추가.

# 스크린샷

### CODE REVIEW

File: BeTime/BeTime/Resource/AppColors.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//
+//  AppColors.swift
+//  BeTime
+//
+//  Created by 쭌이 on 9/13/24.
+//
+
+import UIKit
+
+extension UIColor {
```
Comment (f-lab-barry) [inline BeTime/BeTime/Resource/AppColors.swift:10]: AppColor 보다, common한 이름을 넣거나, 앱이름을 넣는것이 더 좋아보여요

File: BeTime/BeTime/Source/Application/SceneDelegate.swift [hunk 2]
```diff
@@ -15,9 +15,9 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     guard let windowScene = (scene as? UIWindowScene) else { return }
 
     let window = UIWindow(windowScene: windowScene)
-    let mainTabBarController = LaunchScreenViewController()
+    let launchScreen = LaunchScreenViewController()
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Application/SceneDelegate.swift:18]: launchScreenViewController

File: BeTime/BeTime/Source/Scenes/MainTabBarController.swift [hunk 3]
```diff
@@ -18,14 +18,25 @@ final class MainTabBarController: UITabBarController {
...
-    secondViewController.tabBarItem = UITabBarItem(title: "CityList", image: UIImage(systemName: "list.bullet"), selectedImage: UIImage(systemName: "list.bullet"))
+    cityListViewController.tabBarItem = UITabBarItem(
+      title: "CityList",
+      image: UIImage(systemName: "list.bullet"),
+      selectedImage: UIImage(systemName: "list.bullet")
+    )
 
-    self.viewControllers = [firstViewController, secondViewController]
-    self.tabBar.backgroundColor = .lightGray
+    self.viewControllers = [userWeatherViewController, cityListViewController]
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Scenes/MainTabBarController.swift:35]: self.viewControllers = [
  userWeatherViewController,
  cityListViewController,
]

File: BeTime/BeTime/Source/Scenes/MainTabBarController.swift [hunk 4]
```diff
@@ -18,14 +18,25 @@ final class MainTabBarController: UITabBarController {
...
+
+    let appearance = UITabBarAppearance()
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Scenes/MainTabBarController.swift:37]: then 적용할 수 있어보여요

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift [hunk 5]
```diff
@@ -5,10 +5,73 @@
 //  Created by 쭌이 on 9/12/24.
 //
 
+import PinLayout
+import Then
 import UIKit
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:10]: 순서변경필요해요~

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift [hunk 6]
```diff
@@ -5,10 +5,73 @@
...
+  }
+
+  private lazy var addCityButton = UIButton().then {
+    $0.setImage(UIImage(systemName: "plus"), for: .normal)
+    $0.tintColor = .black // 원하는 색상으로 설정
+    $0.backgroundColor = .clear
+    $0.contentMode = .scaleAspectFit
+    $0.addAction(UIAction { _ in
+      print("Plus Button Tapped")
+    }, for: .touchUpInside)
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:26]: `func addActions()` 별도 메서드로 분리하는게 좋을 것 같아요~

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift [hunk 7]
```diff
@@ -5,10 +5,73 @@
...
+  private let tableView = UITableView().then {
+    $0.backgroundColor = .clear
+    $0.separatorStyle = .singleLine
+  }
+
   override func viewDidLoad() {
     super.viewDidLoad()
+    tableView.delegate = self
+    tableView.dataSource = self
+    self.updateUI()
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:38]: ContentView로 분리하는게 좋을 것 같아요~

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift [hunk 8]
```diff
@@ -5,10 +5,73 @@
...
+
+  private func updateUI() {
+    view.backgroundColor = .appBackgroundColor
+    self.setUpSubViews()
+    self.setupConstraint()
+  }
+
+  private func setUpSubViews() {
+    [cityListTitleLabel,addCityButton,tableView
+    ,].forEach {
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:49]:
```suggestion
    [
       cityListTitleLabel,
       addCityButton,
       tableView,
    ].forEach {
```

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift [hunk 9]
```diff
@@ -5,10 +5,73 @@
...
+    cityListTitleLabel.pin.top(100).left(30).width(50).height(30)
+    addCityButton.pin.top(50).right(20).size(40)
+    tableView.pin.below(of: cityListTitleLabel)
+      .marginTop(10).left().right().top().bottom()
+  }
+}
+
+extension CityListViewController: UITableViewDelegate, UITableViewDataSource {
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return 10
```
Comment (f-lab-barry) [inline BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:64]: // NOTE: 다음 커밋에서 수정예정, 네트워크 연결 후 수정예정

File: BeTime/BeTime/Resource/AppFont.swift [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+//  BeTime
+//
+//  Created by 쭌이 on 9/13/24.
+//
+
+import UIKit
+
+extension UIFont {
+
+  static let appTitleFont: UIFont = {
```
Comment (f-lab-barry) [inline BeTime/BeTime/Resource/AppFont.swift:12]: app 은 빼주세요~

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift (hunk 5 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:10]: https://github.com/97JUN/BeTime/pull/14/commits/1243876d57c4ff4476a67a6883fc3da140395145#diff-6f8858b970c0be40fdc7f000cc65e9a454ab1e97ae9ed5d991c67a2948659be6R7

여기서 수정했습니다!

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift (hunk 6 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:26]: https://github.com/97JUN/BeTime/pull/14/commits/1243876d57c4ff4476a67a6883fc3da140395145#diff-6f8858b970c0be40fdc7f000cc65e9a454ab1e97ae9ed5d991c67a2948659be6R166

여기서 수정했습니다!

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift (hunk 7 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:38]: https://github.com/97JUN/BeTime/pull/14/commits/1243876d57c4ff4476a67a6883fc3da140395145#diff-6f8858b970c0be40fdc7f000cc65e9a454ab1e97ae9ed5d991c67a2948659be6R155
여기서 수정했습니다!

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift (hunk 8 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:49]: https://github.com/97JUN/BeTime/pull/14/commits/6d0f7420a2a2b0d0e8030548b80d32caf0d407ae#diff-6f8858b970c0be40fdc7f000cc65e9a454ab1e97ae9ed5d991c67a2948659be6R83
여기서 수정했습니다!

File: BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift (hunk 9 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Scenes/UserCityListScene/CityListViewController.swift:64]: https://github.com/97JUN/BeTime/pull/14/commits/6d0f7420a2a2b0d0e8030548b80d32caf0d407ae#diff-6f8858b970c0be40fdc7f000cc65e9a454ab1e97ae9ed5d991c67a2948659be6R225
여기 반영했습니다!
이후 PR에서 중간과정의 내용은 NOTE로 주석 남기도록 하겠습니다!

File: BeTime/BeTime/Resource/AppColors.swift (hunk 1 above)
Comment (97JUN) [reply BeTime/BeTime/Resource/AppColors.swift:10]: https://github.com/97JUN/BeTime/pull/15/commits/00714913c3e7ea6a8d388cca011dd0ed62a21e03#diff-2dad3e97887b12660ab944f2238aa3f3eb42bcda040b65c870cf31ee6543e8f8L10
여기서 수정했습니다!

File: BeTime/BeTime/Resource/AppFont.swift (hunk 10 above)
Comment (97JUN) [reply BeTime/BeTime/Resource/AppFont.swift:12]: https://github.com/97JUN/BeTime/pull/15/commits/00714913c3e7ea6a8d388cca011dd0ed62a21e03#diff-5fc6f9fc1ac18326a0d4ce3f09074d87e858845ca4fc61b9822a637de74a8375R12
여기서 수정했습니다!

File: BeTime/BeTime/Source/Application/SceneDelegate.swift (hunk 2 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Application/SceneDelegate.swift:18]: https://github.com/97JUN/BeTime/pull/16/commits/37a1b7a595bee96beb5d4595191cf9b06c150a34#diff-311131c35ff6c006c8756f581668d63f697a748eaa66cd23c0e3a739dff6e1d4L18
여기서 수정했습니다

File: BeTime/BeTime/Source/Scenes/MainTabBarController.swift (hunk 3 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Scenes/MainTabBarController.swift:35]: https://github.com/97JUN/BeTime/pull/16/commits/37a1b7a595bee96beb5d4595191cf9b06c150a34#diff-3e190d39daddb207797b03d69d273ae6d1c13b497fc8f3f8b0841b1d9e289951R35
여기서 수정했습니다

File: BeTime/BeTime/Source/Scenes/MainTabBarController.swift (hunk 4 above)
Comment (97JUN) [reply BeTime/BeTime/Source/Scenes/MainTabBarController.swift:37]: https://github.com/97JUN/BeTime/pull/16/commits/37a1b7a595bee96beb5d4595191cf9b06c150a34#diff-3e190d39daddb207797b03d69d273ae6d1c13b497fc8f3f8b0841b1d9e289951R40
여기서 수정했습니다

**Pull request closed without merging**